lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -503,6 +503,12 @@ def add_resources_bundle_targets(file_accessors)
                     configuration.build_settings['CONFIGURATION_BUILD_DIR'] = target.configuration_build_dir('$(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)')
                   end
 
+                  # Set the 'IBSC_MODULE' build settings for iOS resource bundles so that Storyboards and Xibs can load
+                  # classes from the parent module. The setting does not apply to macOS.
+                  unless target.platform == :osx
```

Comment:
does this appear on :watchos and :tvOS? if not lets switch that to 
```ruby
  configuration.build_settings['IBSC_MODULE'] = target.product_module_name if target.platform == :ios
```

Reply:
Just to be fully sure I have checked all platforms and it turns out I was wrong here. The setting is present for all platforms (iOS, macOS, tvOS, watchOS), but it only appears in Xcode if there is a xib for storyboard in the target.

That's why I mistakingly thought it was not present on macOS. I will revert the platform check now.